papermill/utils.py

```diff
@@ -1,11 +1,31 @@
 import os
 import logging
+import warnings
 
 from contextlib import contextmanager
 from functools import wraps
 
 logger = logging.getLogger('papermill.utils')
 
+WARN_ON_ARGS_OVERWRITE = True
+
+def safe_kwargs(caller_args, **callee_args):
+    if WARN_ON_ARGS_OVERWRITE:
+        conflicts = set(caller_args) & set(callee_args)
+        if conflicts:
+            args = format('; '.join(['{}={}'.format(key, value)
+                                     for key, value in callee_args.items()]))
+            msg = 'Callee will overwrite caller''s argument(s): {}'.format(args)
+            warnings.warn(msg)
+    return dict(caller_args, **callee_args)
+
+
+def remove_args(args=None, kwargs=None):

```

Comment:
`def remove_resvered_args(reserved_args=None, user_kwargs=None):`

Reply:
Is this the inverse of `safe_kwargs`?

Reply:
kinda, need to remove args from kwargs intended for funtion but not listed in kwargs or named differently, like timeout/execution_timeout